packages/ic-http-gateway/src/protocol/handler.rs

```diff
@@ -17,6 +17,13 @@ use ic_utils::{
     interfaces::{http_request::HeaderField, HttpRequestCanister},
 };
 
+fn create_err_response(status_code: StatusCode, msg: &str) -> CanisterResponse {
```

Comment:
This isn't the most ergonomic way of creating the `CanisterResponse`, but it means we can guarantee that it will not fail at runtime.

The `Response::builder` will return a `Result` in case some builder methods fail, but we're not even using the ones that can fail so it's an unnecessary cost for us. This also causes a lot of `unwrap` calls because most of the `Response::builder` calls are used in error handling scenarios and we'd have a mess of nested `Result`s if we didn't use `unwrap`.